app/scripts/controllers/transactionDetails.js

```diff
@@ -232,14 +256,16 @@ export function TransactionDetailsCtrl ($scope, $uibModal, $compile, $location,
   /**               Transactions View Body Functions                 **/
   /********************************************************************/
 
-  $scope.viewBodyDetails = function (type, content, headers) {
+  $scope.viewBodyDetails = function (type, content, headers, bodyId) {
     $uibModal.open({
       template: transactionsBodyModal,
       controller: TransactionsBodyModalCtrl,
       windowClass: 'modal-fullview',
       resolve: {
         bodyData: function () {
-          return { type: type, content: content, headers: headers }
+          return {
+            type: type, content: content, headers: headers, transactionId: $routeParams.transactionId, bodyId
```

Comment:
You will also need to pass the current body params (`start, end, bodyLength` and a partial body flag) to the modal. the model need this for future requests that it might make for the response body.